code/modules/shuttle/marine_dropship.dm

```diff
@@ -976,48 +1000,128 @@
 	return ..()
 
 
-/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user)
-	if(!allowed(user))
-		to_chat(user, "<span class='warning'>Access Denied!</span>")
+/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user, datum/tgui/ui)
+	if(!(SSshuttle.getShuttle(shuttleId)))
+		RelinkShuttleId()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ShuttleControl")
+		ui.open()
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_state(mob/user)
+	return GLOB.access_state
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
 		return
+
+	if(action != "selectDestination")
+		return FALSE
+
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!(M.shuttle_flags & GAMEMODE_IMMUNE) && world.time < SSticker.round_start_time + SSticker.mode.deploy_time_lock)
+		to_chat(usr, "<span class='warning'>The engines are still refueling.</span>")
+		return TRUE
+
+	if(!M.can_move_topic(usr))
+		return TRUE
+
+	if(!params["destination"])
+		return TRUE
+
+	if(!(params["destination"] in valid_destinations()))
+		log_admin("[key_name(usr)] may be attempting a href dock exploit on [src] with target location \"[params["destination"]]\"")
+		message_admins("[ADMIN_TPMONTY(usr)] may be attempting a href dock exploit on [src] with target location \"[params["destination"]]\"")
+		return TRUE
+
+	var/previous_status = M.mode
+	log_game("[key_name(usr)] has sent the shuttle [M] to [params["destination"]]")
+
+	switch(SSshuttle.moveShuttle(shuttleId, params["destination"], 1))
+		if(0)
+			if(previous_status != SHUTTLE_IDLE)
+				visible_message("<span class='notice'>Destination updated, recalculating route.</span>")
+			else
+				visible_message("<span class='notice'>Shuttle departing. Please stand away from the doors.</span>")
+			return TRUE
+		if(1)
+			to_chat(usr, "<span class='warning'>Invalid shuttle requested.</span>")
+			return TRUE
+		else
+			to_chat(usr, "<span class='notice'>Unable to comply.</span>")
+			return TRUE
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_data(mob/user)
+	var/list/data = list()
 	var/list/options = valid_destinations()
-	var/obj/docking_port/mobile/marine_dropship/M = SSshuttle.getShuttle(shuttleId)
-	var/dat = "Status: [M ? M.getStatusText() : "*Missing*"]<br><br>"
-	if (M?.hijack_state == HIJACK_STATE_NORMAL)
-		var/destination_found
+	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
+	if(!M)
+		return data //empty but oh well
+
+	data["linked_shuttle_name"] = M.name
+	data["shuttle_status"] = M.getStatusText()
+	for(var/option AS in options)
 		for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
```

Comment:
Nop, you don't need to do AS in for option, because you are not casting option (it's a simple var without a type)
you have to do that for the S var

Reply:
Yeah my late night ape brain thought i put it in the right spot. Welp